source/validate_instruction.cpp

```diff
@@ -131,6 +131,68 @@ spv_result_t CapCheck(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Checks that the number of characters in a string doesn't exceed the limit.
+spv_result_t LimitCheckString(ValidationState_t& _,
+                              const spv_parsed_instruction_t* inst) {
+  // By definition, num_words of an operand is a 16-bit unsigned which will not
+  // exceed the limit of characters for a string literal.
+  // This function checks that a NULL termination must be found before reaching
+  // the end of the string words.
+  for (int i = 0; i < inst->num_operands; ++i) {
+    auto operand = inst->operands[i];
+    if (SPV_OPERAND_TYPE_LITERAL_STRING == operand.type) {
+      bool null_terminated = false;
```

Comment:
I intend to remove this function completely. It seems that if a string is not null-terminated, it will be caught before the validator gets to run. I found this out after manipulating the bytes in my unit test:

```
error: 17: Invalid word count: OpName starting at word 13 says it has 3 words, but found 4 words instead.
```
or this:
```
error: 15: End of input reached while decoding OpName starting at word 13: truncated literal string operand at word offset 2.
```